### PR TITLE
DO NOT MERGE (yet) - chore: refactor static lookup for a proper search index

### PR DESCRIPTION
**Pull request by rozenmd**
**What this PR solves / how to test:**
Quick Win: I basically taught wrangler how to do a "I'm feeling lucky" search

How it works:
- The wrangler user enters `wrangler docs <command>`
- wrangler sends an API request to Algolia's index of https://developers.cloudflare.com/workers/wrangler/
- wrangler sends the user to the first URL in the index that matches the search term

Here's what it looks like in action:

https://user-images.githubusercontent.com/3822106/207370613-eacccce9-ddea-4c9b-82d1-de476ff52fa2.mov

**Reviewer has performed the following, where applicable:**

- [ ] Checked for inclusion of relevant tests
- [ ] Checked for inclusion of a relevant changeset
- [ ] Checked for creation of associated docs updates
- [ ] Manually pulled down the changes and spot-tested

### CODE REVIEW

**Comment by rozenmd** [comment]
Not a secret:
> This is the public API key which can be safely used in your frontend code.This key is usable for search queries and it's also able to list the indices you've got access to.

File: packages/wrangler/src/docs/index.ts [hunk 1]
```diff
@@ -4,90 +4,51 @@ import { logger } from "../logger";
...
-		urlToOpen = "https://developers.cloudflare.com/workers/wrangler/api/";
-	} else if (args.command && isValidParam(args.command)) {
-		//otherwise, they get the wrangler commands page
-		urlToOpen += `#${argToUrlHash[args.command]}`;
-	}
-
 	await printWranglerBanner();
+	if (args.command) {
+		const searchResp = await fetch(
+			`https://${ALGOLIA_APPLICATION_ID}-dsn.algolia.net/1/indexes/developers-cloudflare-wrangler/query`,
```
Comment (GregBrimble) [inline packages/wrangler/src/docs/index.ts:30]: Are we still maintaining our Algolia search instance? I think we moved to Coveo, no?

Comment (rozenmd) [reply packages/wrangler/src/docs/index.ts:30]: This uses Algolia's DocSearch, not our instance

Comment (penalosa) [reply packages/wrangler/src/docs/index.ts:30]: How stable is this? Where are the login credentials etc...?

**Comment by rozenmd** [comment]
Looks like we can't use an external service for this. Gonna look at using lunr.js for offline search instead: https://lunrjs.com/guides/index_prebuilding.html

**Pull request closed without merging**